examples/ProjPlaneCurve.jl

```diff
@@ -0,0 +1,468 @@
+  G = gcd(C.eq.f, D.eq.f)
+  R = parent(C.eq)
+  CC = []
+  if !isone(G)
+     # We divide by the gcd to get curves without common components.
+     F = R(div(C.eq.f, G))
+     H = R(div(D.eq.f, G))
+     push!(CC, ProjPlaneCurve(G))
+  else
+     F = C.eq
+     H = D.eq
+  end
+  Pts = []
+  r, (X, Y) = PolynomialRing(R.R.base_ring, ["X", "Y"])
+  Fa = dehomogenization(r, F, 3)
+  Ha = dehomogenization(r, H, 3)
+  if !isconstant(Fa) && !isconstant(Ha)
+     Ca = AffinePlaneCurveModule.AffinePlaneCurve(Fa)
+     Da = AffinePlaneCurveModule.AffinePlaneCurve(Ha)
+     L = AffinePlaneCurveModule.curve_intersect(Ca, Da)
+     for p in L[2]
+        push!(Pts, Array_to_ProjSpcElem(PP, p.coord))
+     end
+  end
+  rr, (x,) = PolynomialRing(R.R.base_ring, ["x"])
+  phi = hom(R.R, rr, [gens(rr)[1], rr(1), rr(0)])
+  phiF = phi(F)
+  phiH = phi(H)
+  g = gcd(phiF, phiH)
+  f = factor(g)
+  ro = []
+  for h in keys(f.fac)
+     if total_degree(h) == 1
+        f = h//lc(h)
+        push!(ro, -f + gens(rr)[1])
+     end
+  end
+  for y in ro
+     push!(Pts, Oscar.Geometry.ProjSpcElem(PP, [lc(y), R.R.base_ring(1), R.R.base_ring(0)]))
+  end
+  if iszero(evaluate(F, [1, 0, 0])) && iszero(evaluate(H, [1, 0, 0]))
+     push!(Pts, Oscar.Geometry.ProjSpcElem(PP, [R.R.base_ring(1), R.R.base_ring(0), R.R.base_ring(0)]))
+  end
+  return [CC,Pts]
+end
+
+################################################################################
+# without specifying the projective space.
+
+function curve_intersect(C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+   R = parent(C.eq)
+   PP = projective_space(R.R.base_ring, 2)
+   curve_intersect(PP[1], C, D)
+end
+
+################################################################################
+# Compute the singular locus of an affine plane curve by intersecting the
+# derivatives. The points might also be contained in the components.
+
+@doc Markdown.doc"""
+    curve_singular_locus([PP::Oscar.Geometry.ProjSpc{S}], C::ProjPlaneCurve{S}) where S <: FieldElem
+
+Return the reduced singular locus of `C` as a list whose first element is the projective plane curve consisting of the singular components of `C` (if any), and the second element is the list of the singular points of the reduction of `C` (the points are in `PP` if specified, or in a new projective space otherwise). The singular component might not contain any point over the considered field.
+"""
+function curve_singular_locus(PP::Oscar.Geometry.ProjSpc{S}, C::ProjPlaneCurve{S}) where S <: FieldElem
+  D = reduction(C)
+  Pts = Array{Oscar.Geometry.ProjSpcElem, 1}()
+  CC = Array{ProjPlaneCurve, 1}()
+  # The components with multiplicity > 1 are singular
+  f = []
+  for (h, c) in C.components
+     if c != 1
+        push!(f, h.eq)
+     end
+  end
+  if !isempty(f)
+     g = prod(f)
+     push!(CC, ProjPlaneCurve(g))
+  end
+  J = jacobi_ideal(D)
+  FX = gens(J)[1]
+  FY = gens(J)[2]
+  FZ = gens(J)[3]
```

Comment:
I wonder if this would work? (Not tested, and I don't ask you to change this in this PR, really just wondering)
```suggestion
  FX, FY, FZ = gens(J)
```